### DISCUSSION TITLE
Common Gateway API installation experience

### DISCUSSION THREAD

**Discussion by jpeach** [kubernetes-sigs/gateway-api#934] [Ideas]
As more project implement Gateway API, there are some cross-project concerns about how to configure and manage the common API layer.

Basically every implementation will need to
* ensure the correct version of the Gateway CRDs is installed
* ensure the Gateway validating webhook is installed

A number of implementations use Helm charts as their canonical installation method, but also have had bad experiences trying to manage CRDs in Helm. The Helm project disclaims upgrading CRDs in charts, though it does provide some general CRD guidance.

WG consensus is that `kubectl install` is currently the best way to provide the CRDs, but there is no obvious YAML to get the varidation webhook.

What we would like is
* some sort of standardized installation experience across implementations
* some sort of bundling so that CRDs, webhooks and other WG controllers can be installed together
* a reasonable story for upgrading the components in the bundle

Lets use this discussion to collect experiences and opinions, then we can filter out some issues to work on.

Ref:
* #930
* #629 
* #502 
* #10 
* https://helm.sh/docs/chart_best_practices/custom_resource_definitions/

**Comment by jpeach** [comment]
Gateway API installation for different controllers

| Controller | installation |
| --- | --- |
| Contour | https://projectcontour.io/guides/gateway-api/ |

**Reply by Jeff-Apple** [reply]
| Controller | Installation |
|-------------|-------------|
| Consul API Gateway | [https://www.consul.io/docs/api-gateway/api-gateway-usage](https://www.consul.io/docs/api-gateway/api-gateway-usage)

**Reply by howardjohn** [reply]
Controller | Installation
-- | --
Istio | https://istio.io/latest/docs/tasks/traffic-management/ingress/gateway-api/

Note - this assumes you already have the Istio control plane installed
